src/pyrenew/process/iidrandomsequence.py

```diff
@@ -0,0 +1,150 @@
+# -*- coding: utf-8 -*-
+# numpydoc ignore=GL08
+
+import numpyro.distributions as dist
+from numpyro.contrib.control_flow import scan
+
+from pyrenew.metaclass import DistributionalRV, RandomVariable, SampledValue
+
+
+class IIDRandomSequence(RandomVariable):
+    """
+    Class for constructing random sequence of
+    independent and identically distributed elements
+    given an arbitrary RandomVariable representing
+    those elements.
+    """
+
+    def __init__(
+        self,
+        name: str,
```

Comment:
I don't think this `name` actually gets used? Wouldn't you need a `numpyro.deterministic` call in `sample` to do that?

Reply:
True, I was including it on the grounds that we might in the future move toward all RVs being mandatorily named / optionally recorded. But happy to remove for now if you want.

Reply:
I think I prefer removing, but am somewhat interested in an optional `name` and/or optional `record`.

Reply:
Will remove.

Reply:
related: https://github.com/CDCgov/PyRenew/issues/402

Reply:
Removed in 7cc5377

Reply:
@dylanhmorris You removed names in a different section than the one referenced here.

Reply:
addressed in 27f0c52